src/components/Output/custom-els/RangeInput/index.ts

```diff
@@ -0,0 +1,94 @@
+import './styles.css';
+
+const EVENTS = [
+  'input',
+  'change',
+  'focus',
+  'blur',
+];
+
+const PROPERTIES = [
+  'name',
+  'value',
+  'min',
+  'max',
+  'step',
+];
+
+interface RangeInputElement {
+  value: string;
+  min?: string;
+  max?: string;
+  step?: string;
+  precision?: number | string;
+}
+
+class RangeInputElement extends HTMLElement {
+  private _input = document.createElement('input');
+  private _valueDisplayWrapper = document.createElement('aside');
+  private _valueDisplay = document.createElement('span');
+
+  constructor() {
+    super();
+    this._input.type = 'range';
+    this._update = this._update.bind(this);
+    this._handleEvent = this._handleEvent.bind(this);
+
+    for (const event of EVENTS) {
+      this._input.addEventListener(event, this._handleEvent, true);
+    }
+
+    for (const property of PROPERTIES) {
+      Object.defineProperty(this, property, {
+        configurable: true,
+        get() {
+          return this._input[property];
+        },
+        set(value) {
+          this._input[property] = value;
+          this._update();
+        },
+      });
+    }
+  }
+
+  connectedCallback() {
+    if (this._input.parentNode !== this) {
+      this.appendChild(this._input);
+      this._valueDisplayWrapper.appendChild(this._valueDisplay);
+      this.appendChild(this._valueDisplayWrapper);
+    }
+  }
+
+  private _handleEvent(event: Event) {
+    this._update();
+    event.stopImmediatePropagation();
+    const retargetted = new (event.constructor as typeof Event)(event.type, event);
```

Comment:
Does `... as Constructor<Event>` work?

Reply:
But also, is this redirection necessary?

Reply:
it's the same approach Jake used elsewhere, was the only thing I could get to work.  I could just use the Event constructor maybe? But there are multiple event types.

Reply:
`new Event` is fine here since all the events being retargeted are just regular events.